site/content/entry/yuzu-progress-report-apr-2022/index.md

```diff
@@ -0,0 +1,333 @@
+When the game closes the client port, the service closes the server port, and everything is shut down.
+
+The issue with our previous implementation was that yuzu wasn't properly tracking all the `KServerPort` and `KServerSession` objects for each service.
+And because of this, the services weren't properly getting closed and they in turn were causing further issues.
+
+This originally worked fine, but was regressed when we migrated guest kernel objects to emulated guest memory, as we mentioned previously.
+bunnei figured out the issue and quickly {{< gh-hovercard "8165" "reimplemented how we track these kernel objects" >}}.
+
+By having a single place where we can register/unregister open ports and sessions, we can now keep better track of these kernel objects.
+And by ensuring that they are closed when we tear down all services and kernel, we get much better emulation shutdown behaviour.
+
+## Input changes and general bugfixes
+
+If the user sets a very high DPI value for their mouse while using mouse panning, the cursor may be able to escape the rendering window.
+[IamSanjid](https://github.com/IamSanjid) implemented the {{< gh-hovercard "8170" "required fixes" >}}, including better centering timings to solve this issue. Thanks!
+
+[german77](https://github.com/german77) has several fixes ready for us.
+
+Let’s begin with an interesting one.
+yuzu’s screenshot capture feature allows an easy way to save moments at the resolution the scaler is currently set at.
+The hotkey for capture could be spammed, leading yuzu to a crash if several requests for capture were sent, this could be aggravated if the rendering resolution was set to a high value.
+To solve this, yuzu now {{< gh-hovercard "8192" "ignores new requests while a capture is being processed" >}}, and prints a warning in the log.
+
+There’s always room for improvements in emulation, as nothing is ever truly complete. This time, german77 focuses on inaccuracies found on our input emulation.
+
+{{< gh-hovercard "8222" "`IsSixAxisSensorFusionEnabled` is implemented" >}} by reverse engineering all sixaxis functions, and it was verified by comparing with unit test homebrew results done on the Switch.
+This should potentially improve motion accuracy.
+
+The [HID](https://en.wikipedia.org/wiki/Human_interface_device) service in charge of handling input commands, among other things, used to operate by copying its assigned shared memory and reporting back the changes.
+This leads to mismatches or delays in the input process, and can potentially make games read completely wrong data.
+
+Of course, this is not ideal at all, so german77 {{< gh-hovercard "8229" "gets rid of the memory duplication" >}} and uses the ever magical * pointers to access the shared memory directly instead.
+This can fix bugs on countless games, with the biggest example being the `Pokémon: Let’s Go` games having a hard time detecting controllers.
```

Comment:
```suggestion
This can fix bugs affecting countless games, with the biggest example being the `Pokémon: Let’s Go` games having a hard time detecting controllers.
```